src/openms/source/ANALYSIS/ID/SimpleSearchEngineAlgorithm.cpp

```diff
@@ -0,0 +1,654 @@
+    search_parameters.missed_cleavages = peptide_missed_cleavages;
+    search_parameters.fragment_mass_tolerance = fragment_mass_tolerance;
+    search_parameters.precursor_mass_tolerance = precursor_mass_tolerance;
+    search_parameters.precursor_mass_tolerance_ppm = precursor_mass_tolerance_unit_ppm == "ppm" ? true : false;
+    search_parameters.fragment_mass_tolerance_ppm = fragment_mass_tolerance_unit_ppm == "ppm" ? true : false;
+    search_parameters.digestion_enzyme = *ProteaseDB::getInstance()->getEnzyme(enzyme);
+    protein_ids[0].setSearchParameters(search_parameters);
+  }
+
+  SimpleSearchEngineAlgorithm::ExitCodes SimpleSearchEngineAlgorithm::search(const String& in_mzML, const String& in_db, vector<ProteinIdentification>& protein_ids, vector<PeptideIdentification>& peptide_ids) const
+  {
+    boost::regex peptide_motif_regex(peptide_motif_);
+
+    bool precursor_mass_tolerance_unit_ppm = (precursor_mass_tolerance_unit_ == "ppm");
+    bool fragment_mass_tolerance_unit_ppm = (fragment_mass_tolerance_unit_ == "ppm");
+
+    set<String> fixed_unique(modifications_fixed_.begin(), modifications_fixed_.end());
+
+    if (fixed_unique.size() != modifications_fixed_.size())
+    {
+      cout << "duplicate fixed modification provided." << endl;
+      return ExitCodes::ILLEGAL_PARAMETERS;
+    }
+
+    set<String> var_unique(modifications_variable_.begin(), modifications_variable_.end());
+    if (var_unique.size() != modifications_variable_.size())
+    {
+      cout << "duplicate variable modification provided." << endl;
+      return ExitCodes::ILLEGAL_PARAMETERS;
+    }
+
+    vector<ResidueModification> fixed_modifications = getModifications_(modifications_fixed_);
+    vector<ResidueModification> variable_modifications = getModifications_(modifications_variable_);
+
+    // load MS2 map
+    PeakMap spectra;
+    MzMLFile f;
+    //f.setLogType(log_type_);
+
+    PeakFileOptions options;
+    options.clearMSLevels();
+    options.addMSLevel(2);
+    f.getOptions() = options;
+    f.load(in_mzML, spectra);
+    spectra.sortSpectra(true);
+
+    startProgress(0, 1, "Filtering spectra...");
+    preprocessSpectra_(spectra, fragment_mass_tolerance_, fragment_mass_tolerance_unit_ppm);
+    endProgress();
+
+    // build multimap of precursor mass to scan index
+    multimap<double, Size> multimap_mass_2_scan_index;
+    for (PeakMap::ConstIterator s_it = spectra.begin(); s_it != spectra.end(); ++s_it)
+    {
+      int scan_index = s_it - spectra.begin();
+      vector<Precursor> precursor = s_it->getPrecursors();
+
+      // there should only one precursor and MS2 should contain at least a few peaks to be considered (e.g. at least for every AA in the peptide)
+      if (precursor.size() == 1 && s_it->size() >= peptide_min_size_)
+      {
+        int precursor_charge = precursor[0].getCharge();
+
+        if (precursor_charge < precursor_min_charge_ 
+         || precursor_charge > precursor_max_charge_)
```

Comment:
395 / 396: 

/.../OpenMS/src/openms/source/ANALYSIS/ID/SimpleSearchEngineAlgorithm.cpp:395:32: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]


the question is whether the charge can actually be negative here - do you support negative charges?

Reply:
no I am even not sure if the TheoreticalSpectraGenerators do so this is always >=1